core/src/main/java/com/linecorp/armeria/client/http/HttpClientPipelineConfigurator.java

```diff
@@ -114,20 +116,13 @@
     private InetSocketAddress remoteAddress;
 
     HttpClientPipelineConfigurator(SessionProtocol sessionProtocol, SessionOptions options) {
-        switch (sessionProtocol) {
-        case HTTP:
-        case HTTPS:
+        if (sessionProtocol == HTTP || sessionProtocol == HTTPS) {
             httpPreference = HttpPreference.HTTP2_PREFERRED;
-            break;
-        case H1:
-        case H1C:
+        } else if (sessionProtocol == H1 || sessionProtocol == H1C) {
             httpPreference = HttpPreference.HTTP1_REQUIRED;
-            break;
-        case H2:
-        case H2C:
+        } else if (sessionProtocol == H2 || sessionProtocol == H2C) {
             httpPreference = HttpPreference.HTTP2_REQUIRED;
-            break;
-        default:
+        } else {
```

Comment:
Yeah, it's probably gonna be the day we hit Armeria 1.0 - having generic enough API that can handle two very different reqres protocols and various serialization formats!